bobtask/verify.go

```diff
@@ -32,9 +35,36 @@ func (t *Task) verifyBefore() (err error) {
 			}
 		}
 	}
+
+	if t.target != nil && t.target.Type() == target.Path {
+		for _, path := range t.target.Paths() {
+			if verifyTargetPath(path) {
+				return usererror.Wrap(fmt.Errorf("invalid target `%s` for task `%s`", path, t.name))
+			}
+		}
+	}
+
 	return nil
 }
 
+// verifyTargetPath reports whether the final component of path is "." or ".."
+func verifyTargetPath(path string) bool {
```

Comment:
The function definition/usage is a bit misleading. If `verifyTargetPath() returns true we should not expect to throw an error. Maybe rename to isValidFilesystemTarget() and adjust it's behavior.

Also: Does it make sense to only validate the last part of the path? Shouldn't it check each subpart of the path?
I can imagine targets like those:
* my/../strange/target
* ../target

Would also be nice to have a unit test for this function.

Reply:
Updated the tests to cover also /../ parts